src/cpu.rs

```diff
@@ -25,116 +25,90 @@ impl Cpu {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::CNZ(addr) => match self.cnz(addr) {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::CP(addr) => match self.cp(addr) {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::CM(addr) => match self.cm(addr) {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::CPE(addr) => match self.cpe(addr) {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::CPO(addr) => match self.cpo(addr) {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RET => self.ret(),
-            Instruction::RC => match self.rc() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RNC => match self.rnc() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RZ => match self.rz() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RNZ => match self.rnz() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RP => match self.rp() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RM => match self.rm() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RPE => match self.rpe() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RPO => match self.rpo() {
-                None => self.pc.wrapping_add(instruction.size()),
-                Some(next_pc) => next_pc,
-            },
-            Instruction::RST(addr) => self.rst(addr),
+        // possibly rename this to something more appropriate if other
+        // instructions will use this
+        macro_rules! unconditional_instruction {
+            ($F:ident, $P:ident) => {
+                (self.$F($P), instruction.cycles())
+            };
+            ($F:ident) => {
+                (self.$F(), instruction.cycles())
+            };
+        }
```

Comment:
As the comment says, if some of the other instructions' actions would apply to this, this macro will probably need to be renamed to something more appropriate.